frontend/src/metabase/components/PaginationControls/PaginationControls.tsx

```diff
@@ -0,0 +1,72 @@
+import { t } from "ttag";
+
+import Button from "metabase/core/components/Button";
+import { Group, Text } from "metabase/ui";
+
+const isLastPage = (pageIndex: number, pageSize: number, total: number) =>
+  pageIndex === Math.ceil(total / pageSize) - 1;
+
+export type PaginationControlsProps = {
+  page: number;
+  pageSize: number;
+  itemsLength: number;
+  total?: number;
+  showTotal?: boolean;
+  onNextPage?: (() => void) | null;
+  onPreviousPage?: (() => void) | null;
+};
+
+export const PaginationControls = ({
+  page,
+  pageSize,
+  itemsLength,
+  total,
+  showTotal = false,
+  onNextPage,
+  onPreviousPage,
+}: PaginationControlsProps) => {
+  const isSinglePage = total !== undefined && total <= pageSize;
+
+  if (isSinglePage) {
+    return null;
+  }
+
+  const isPreviousDisabled = page === 0;
+  const isNextDisabled =
+    total != null ? isLastPage(page, pageSize, total) : !onNextPage;
+
+  return (
+    <Group align="center" fw="bold" aria-label="pagination">
+      <Text span mr="sm">
```

Comment:
I think `mr` is OK since if we ever do attempt to introduce proper RTL compatibility, we'll go through and replace all the `mr`s with `me`s. `me` is supported in [Mantine v7](https://mantine.dev/styles/style-props/), which we'll hopefully be able to switch to soon